prov/util/src/util_buf.c

```diff
@@ -115,9 +115,9 @@ int util_buf_grow(struct util_buf_pool *pool)
 		if (!new_table)
 			goto err3;
 		pool->regions_table = new_table;
-		pool->regions_table[pool->regions_cnt] = buf_region;
-		pool->regions_cnt++;
 	}
+	pool->regions_table[pool->regions_cnt] = buf_region;
+	pool->regions_cnt++;
```

Comment:
The commit message seems to be lacking the reason for this change.  This looks like a fix to a bug.

Reply:
> Just FYI, this broke mrail. It looks like cq_readfrom() doesn't return the correct address/index anymore. I unfortunately didn't have the time to dig deeper... :-(

@yburette Thanks for testing this!
Sorry, there were some problems with util/buf that causes memory corruption  -- it was the night when I was implementing and pushing it for review :).
Could you take a look again, please?